api/net/Areas/Admin/Models/DataSource/DataSourceModel.cs

```diff
@@ -174,9 +174,9 @@ public DataSourceModel(Entities.DataSource entity, JsonSerializerOptions options
         this.Topic = entity.Topic;
         this.ParentId = entity.ParentId;
         this.Connection = JsonSerializer.Deserialize<Dictionary<string, object>>(entity.Connection, options) ?? new Dictionary<string, object>();
-        this.LastRanOn = entity.LastRanOn;
         this.RetryLimit = entity.RetryLimit;
-        this.FailedAttempts = entity.FailedAttempts;
+        this.LastRanOn = entity.DataService?.LastRanOn;
```

Comment:
Needed to separate the properties that are updated by the ingestion services so that they don't cause optimistic concurrency isssues.